app/models/criterion.rb

```diff
@@ -159,4 +162,37 @@ def self.update_assigned_groups_counts(assignment, criterion_ids_by_type = nil)
     #{"AND id IN (#{checkbox_criterion_ids_str})" unless checkbox_criterion_ids_str.empty?}
     UPDATE_SQL
   end
+
+  def replace_marks
+    m = []
+    # results with specific assignment
+    results = Result
+                .joins(submission: :grouping)
+                .where(groupings: {assignment_id: self.assignment_id})
+    if self.ta_visible_changed?
+      # existing marks should be deleted
+      results.each do |r|
+        if !r.is_a_review? # filter results that are not peer reviews
```

Comment:
Use `unless <condition>` instead of `if !<condition>` when there is no else branch. This also happens below.